packages/doenetml-worker-rust/lib-doenetml-core/src/components/types.rs

```diff
@@ -98,6 +99,51 @@ impl<T: IntoGraphNode> From<T> for ComponentIdx {
     }
 }
 
+/// The index of the component in `Core.components`
```

Comment:
Need to correct docstring.

It looks like `StringIdx` is used only in `ContentRef` and it can be converted into a `GraphNode`. I don't believe we need to derive `Hash` on it.

I can't put a comment on line 52 as not part of this changeset. But, we should update the docstring to refer to the index of the component in `DocumentStructure.components`.

Reply:
It turns out that you use `Hash` in `get_flat_dast_updates`

Reply:
Yes, we need to have `ComponentIdx` hashed for the current implementation of flat dast updates, but not `StringIdx`.

Reply:
Still need to correct this docstring and the docstirng in line 52.